src/coronawhy_vt/utils/local_processing.py

```diff
@@ -0,0 +1,371 @@
+# -*- coding: utf-8 -*-
+
+"""Temporary storage for utils used in the processing of local data structures."""
+# TODO function names are in camelcase suggest refactoring to conform to PEP-8
+import json
+import os
+from typing import Callable
+
+import dateutil.parser as parser
+import numpy as np
+import pandas as pd
+import pylab
+import spacy
+
+# TODO All of these functions open a file and do not close it, putting them in the same class would solve some issues
+# TODO lowercase?
+
+
+def make_ngram_map_and_list(filename: str) -> (list, dict):
+    """Given the path of a file of tokens, replace every " " with "_".
+
+    :param filename: The path of the file
+
+    :return (names, ngram_map): (Original tokens with spaces instead of underscores
+        {names: names_with_space_repaced_by_underscores})
+    """
+    ngram_map = {}
+    names = []
+    all_names = []
+
+    file = open(filename).readlines()
+    for line in file:
+        # TODO this is meant to remove the last newline character but can remove the last character of the last line if
+        # file does not end with newline recommend rstrip()
+        all_names.append(line[:-1])
+
+    for name in all_names:
+        # if(len(name) > 5):        # Why this restriction? to filter out unnecessary words
+        if " " in name:
+            # TODO Why are we only limiting the number of tokens to 4 for replacing spaces? same as above basically
+            # if((len(name.split(" ")) > 1) and len(name.split(" ")) < 4):
+            newname = name.replace(" ", "_").lower()
+            ngram_map[name.lower()] = newname
+            names.append(newname)
+        else:
+            names.append(name.lower())
+    names = np.unique(names)
+    return names, ngram_map
+
+
+# TODO Pull the next 6 functions into a class?
+# These functions determine what blocks are pulled from the paper for matching
+def title_blocks(paper: dict) -> list:
+    """Retrieve title block of a paper in json form.
+
+    :param paper (json): The paper to be parsed
+
+    :return: Lines of text in the paper title
+    """
+    return([{'text': paper['metadata']['title']}])
+
+
+def abstract_blocks(paper: dict) -> list:
+    """Retrieve abstract block of a paper in json form.
+
+    :param paper: The paper to be parsed
+
+    :return: Lines of text in the paper abstract
+    """
+    return(paper['abstract'])
+
+
+def body_blocks(paper: dict) -> list:
```

Comment:
Is this function really necessary? It adds 9 lines where it might just be more appropriate to write `paper['body_text']` anywhere you would call it. This philosophy comes from Jack Diedrich (see his excellent talk at https://www.youtube.com/watch?v=o9pEzgHorH0)

Reply:
Yeah, my thought too although I wanted this PR to just be a move of the original functions into proper package and reproducible test form. I'll open a new PR for streamlining the methods.